chain/stmgr/call.go

```diff
@@ -108,7 +108,10 @@ func (sm *StateManager) CallWithGas(ctx context.Context, msg *types.Message, pri
 		ts = sm.cs.GetHeaviestTipSet()
 	}
 
-	state := ts.ParentState()
+	state, _, err := sm.TipSetState(ctx, ts)
```

Comment:
This is the important fix in GasLimit estimation. If a message was already included but not executed we were not taking it into account which could cause out of gas or for example faucet issues.